exporter/datadogexporter/translate_traces_test.go

```diff
@@ -984,3 +984,31 @@ func TestStatsAggregations(t *testing.T) {
 
 	assert.Equal(t, "test-version", statsVersionTag.Value)
 }
+
+// ensure that sanitization  of trace payloads occurs
+func TestSanitization(t *testing.T) {
+	calculator := newSublayerCalculator()
+
+	traces := pdata.NewTraces()
+	traces.ResourceSpans().Resize(1)
+	rs := traces.ResourceSpans().At(0)
+	resource := rs.Resource()
+	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		"deployment.environment": pdata.NewAttributeValueString("UpperCase"),
+	})
+	rs.InstrumentationLibrarySpans().Resize(1)
+	ilss := rs.InstrumentationLibrarySpans().At(0)
+	instrumentationLibrary := ilss.InstrumentationLibrary()
+	instrumentationLibrary.SetName("flash")
+	instrumentationLibrary.SetVersion("v1")
+	ilss.Spans().Resize(1)
+
+	outputTraces, _ := convertToDatadogTd(traces, calculator, &config.Config{})
+	aggregatedTraces := aggregateTracePayloadsByEnv(outputTraces)
+
+	obfuscator := obfuscate.NewObfuscator(obfuscatorConfig)
+	obfuscatePayload(obfuscator, aggregatedTraces)
+	assert.Equal(t, 1, len(aggregatedTraces))
+
+	assert.Equal(t, "uppercase", aggregatedTraces[0].Env)
```

Comment:
What about adding some more tests to cover the other cases? Not saying that we get *all* of the test cases from here https://github.com/DataDog/datadog-agent/blob/c87e93a75b1fc97f0691faf78ae8eb2c280d6f55/pkg/trace/traceutil/normalize_test.go#L12 but a couple more for common issues would be nice

Reply:
makes sense, will do